test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test how the wallet deals with TRUC transactions"""
+
+from decimal import Decimal, getcontext
+
+from test_framework.authproxy import JSONRPCException
+from test_framework.messages import (
+    COIN,
+    CTransaction,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_RETURN
+)
+
+from test_framework.script_util import bulk_vout
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_greater_than,
+    assert_raises_rpc_error,
+)
+
+from test_framework.mempool_util import (
+    TRUC_MAX_VSIZE,
+    TRUC_CHILD_MAX_VSIZE,
+)
+
+def cleanup(func):
+    def wrapper(self, *args):
+        try:
+            self.generate(self.nodes[0], 1)
+            func(self, *args)
+        finally:
+            self.sync_mempools()
+            self.generate(self.nodes[0], 1)
+            try:
+                self.alice.sendall([self.charlie.getnewaddress()])
+            except JSONRPCException as e:
+                assert "Total value of UTXO pool too low to pay for transaction" in e.error['message']
+            try:
+                self.bob.sendall([self.charlie.getnewaddress()])
+            except JSONRPCException as e:
+                assert "Total value of UTXO pool too low to pay for transaction" in e.error['message']
+            self.sync_mempools()
+            self.generate(self.nodes[0], 1)
+            assert_equal(0, self.alice.getbalances()["mine"]["untrusted_pending"])
+            assert_equal(0, self.bob.getbalances()["mine"]["untrusted_pending"])
+            assert_equal(0, self.alice.getbalances()["mine"]["trusted"])
+            assert_equal(0, self.bob.getbalances()["mine"]["trusted"])
+            assert_equal(0, self.alice.getbalances()["mine"]["immature"])
+            assert_equal(0, self.bob.getbalances()["mine"]["immature"])
+            assert_equal(self.alice.getrawmempool(), [])
+            assert_equal(self.bob.getrawmempool(), [])
+
+    return wrapper
+
+class WalletV3Test(BitcoinTestFramework):
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def set_test_params(self):
+        getcontext().prec=10
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        # whitelist peers to speed up tx relay / mempool sync
+        self.noban_tx_relay = True
```

Comment:
In https://github.com/bitcoin/bitcoin/commit/f4d209c0e94fad8435215abba3e61d79585d0d12 "test: add truc wallet tests"

I don't think this is required because there is only one node in the test.